src/main/java/edu/unc/lib/boxc/migration/cdm/services/FindingAidReportService.java

```diff
@@ -201,34 +202,30 @@ private Integer countRecords(String query) {
     }
 
     /**
-     * List of potential collection id values
+     * Map of potential collection id values and the fields they come from
      */
-    private List<String> listPotentialCollectionIds() {
+    private Map<String, String> listPotentialCollectionIds() {
         fieldService.validateFieldsFile(project);
         CdmFieldInfo fieldInfo = fieldService.loadFieldsFromProject(project);
         List<String> exportFields = fieldInfo.listAllExportFields();
 
-        List<String> potentialCollectionIds = new ArrayList<>();
+        Map<String, String> potentialCollectionIds = new HashMap<>();
         // only collect the first 10 potential ids
-        int i = 0;
         try (Connection conn = indexService.openDbConnection()) {
             var stmt = conn.createStatement();
             for (String field : exportFields) {
-                ResultSet rs = stmt.executeQuery(" select '" + field + "' from " + CdmIndexService.TB_NAME
-                        + " where '" + field + "' like " + "'^[A-Za-z0-9]{5}-?z?$'");
+                ResultSet rs = stmt.executeQuery(" select " + field + " from " + CdmIndexService.TB_NAME
+                        + " where " + field + " like " + "'^[A-Za-z0-9]{5}-?z?$' limit 10");
                 while (rs.next()) {
-                    if (!rs.getString(1).isBlank() && i < 10) {
-                        potentialCollectionIds.add(rs.getString(1));
-                        i++;
-                    }
+                    potentialCollectionIds.put(field, rs.getString(1));
```

Comment:
I'm not sure if this is intentional, but now we will only get one value per field (whatever one is found last). We could pass back `Map<String, List<String>>` so that all the values get returned. Or keep adding the values to the existing string for that field with some kind of separator like a newline or ", ", but that might be better left up to the presentation code. If it's okay to only return one value, then the limit should probably be adjusted to 1